File: TEST data for large repo/repo-Copy5/Blueprism/BluePrism.DigitalWorker/EnvironmentFunctions/GetResourceNameFunction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using BluePrism.AutomateAppCore;
using BluePrism.AutomateAppCore.Resources;
using BluePrism.AutomateProcessCore;
using BluePrism.Server.Domain.Models;

namespace BluePrism.DigitalWorker.EnvironmentFunctions
{    
    public class GetResourceNameFunction : GetResourceNameFunctionBase
    {
        private readonly Func<DigitalWorkerContext> _digitalWorkerContextFactory;

        public GetResourceNameFunction(Func<DigitalWorkerContext> digitalWorkerContextFactory)
        {
            _digitalWorkerContextFactory = digitalWorkerContextFactory;
        }

        protected override clsProcessValue InnerEvaluate(IList<clsProcessValue> parameters, clsProcess process)
        {
            if (parameters.Any())            
                throw GetResourceNameFunctionShouldNotHaveAnyParamsException;

            return _digitalWorkerContextFactory().StartUpOptions.Name.FullName;
        }
    }
}

File: TEST data for large repo/Repo_copy/Blueprism/BluePrism.Automate/AutomateAppCore/EnvironmentFunctions/GetResourceNameFunction.vb
Imports BluePrism.AutomateProcessCore

Namespace EnvironmentFunctions

    Public Class GetResourceNameFunction : Inherits GetResourceNameFunctionBase
        
        Private ReadOnly mServerFactory As Func(Of IServer)

        Public Sub New(serverFactory As Func(Of IServer))
            mServerFactory = serverFactory
        End Sub

        Protected Overrides Function InnerEvaluate(parameters As IList(Of clsProcessValue), process As clsProcess) As clsProcessValue
            If parameters.Any() Then
                Throw GetResourceNameFunctionShouldNotHaveAnyParamsException()
            End If

            If process.Session Is Nothing Then
                Throw New clsFunctionException(My.Resources.GetResourceNameFunction_NotRunning)
            End If

            Dim resourceName = mServerFactory().GetResourceNameFromSessionId(process.Session.ID)

            Return New clsProcessValue(DataType.text, resourceName, True)
        End Function
    End Class
End NameSpace